hack/util.sh

```diff
@@ -609,9 +617,9 @@ function install_router {
                  --cert="${MASTER_CONFIG_DIR}/router.crt" --key="${MASTER_CONFIG_DIR}/router.key"
             cat "${MASTER_CONFIG_DIR}/router.crt" "${MASTER_CONFIG_DIR}/router.key" \
                 "${MASTER_CONFIG_DIR}/ca.crt" > "${MASTER_CONFIG_DIR}/router.pem"
-            ROUTER_DEFAULT_CERT="--default-cert=${MASTER_CONFIG_DIR}/router.pem"
+            router_args+=("--default-cert='${MASTER_CONFIG_DIR}/router.pem'")
```

Comment:
You want to quote to avoid shell word-splitting, but you don't want to pass literal quotes to the command AFAIK.  What you really need is to omit the single-quotes here and use `"${router_args[@]}"` below:

```console
$ foo=( --foo --bar ); foo+=( "--baz='quux xyzzy'" )
$ printf ':%s: ' ${foo[*]}; echo
:--foo: :--bar: :--baz='quux: :xyzzy':
```

It's passing the quote marks to the command *and* splitting `--baz='quux xyzzy'` .

```console
$ printf ':%s: ' "${foo[@]}"; echo
:--foo: :--bar: :--baz='quux xyzzy':
```

It's passing `--baz='quux xyzzy'` as a single word, but it still includes the quote marks.

```console
$ foo=( --foo --bar ); foo+=( "--baz=quux xyzzy" )
$ printf ':%s: ' ${foo[*]}; echo
:--foo: :--bar: :--baz=quux: :xyzzy:
```

We got rid of the quote marks, but we're back to splitting `--baz=quux xyzzy`.

```console
$ printf ':%s: ' "${foo[@]}"; echo
:--foo: :--bar: :--baz=quux xyzzy:
```

This is what we want.

Does that make sense?